src/realm/sync/client.cpp

```diff
@@ -780,44 +786,44 @@ void SessionImpl::process_pending_flx_bootstrap()
     VersionInfo new_version;
     DownloadCursor download_cursor;
     int64_t query_version = -1;
-    try {
-        while (bootstrap_store->has_pending()) {
-            auto pending_batch = bootstrap_store->peek_pending(batch_size_in_bytes);
-            if (!pending_batch.progress) {
-                logger.info("Incomplete pending bootstrap found for query version %1", pending_batch.query_version);
-                bootstrap_store->clear();
-                return;
-            }
+    while (bootstrap_store->has_pending()) {
```

Comment:
This is actually the same as before just left-shifted by one since we now do the try catch block outside of this function.